mex/common/sinks/registry.py

```diff
@@ -0,0 +1,79 @@
+from collections.abc import Generator, Iterable
+from itertools import tee
+from typing import Final
+
+from mex.common.models import AnyExtractedModel
+from mex.common.settings import BaseSettings
+from mex.common.sinks.backend_api import BackendApiSink
+from mex.common.sinks.base import BaseSink
+from mex.common.sinks.ndjson import NdjsonSink
+from mex.common.types import Identifier, Sink
+
+_SINK_REGISTRY: Final[dict[Sink, type["BaseSink"]]] = {}
+
+
+class _MultiSink(BaseSink):
+    """Sink to load models to multiple sinks simultaneously."""
+
+    # This class is private because it should only be acquired by calling `get_sink`.
+
+    _sinks: list[BaseSink] = []
+
+    def __init__(self) -> None:
+        """Instantiate the multi sink singleton."""
+        settings = BaseSettings.get()
+        for sink in settings.sink:
+            if sink in _SINK_REGISTRY:
+                sink_cls = _SINK_REGISTRY[sink]
+                self._sinks.append(sink_cls.get())
+            else:
+                msg = f"Sink function not implemented: {sink}"
+                raise RuntimeError(msg)
+
+    def close(self) -> None:
+        """Close all underlying sinks."""
+        for sink in self._sinks:
+            sink.close()
+
+    def load(
+        self,
+        models: Iterable[AnyExtractedModel],
+    ) -> Generator[Identifier, None, None]:
+        """Load models to multiple sinks simultaneously."""
+        for sink, model_gen in zip(
+            self._sinks, tee(models, len(self._sinks)), strict=False
```

Comment:
why not strict? Don't we always want as many model generators as we have `self._sinks`?

Reply:
good catch. this was a ruff autofix for https://docs.astral.sh/ruff/rules/zip-without-explicit-strict - but not actually what we want here. fixed -> 08a4502